stablehlo/dialect/Version.cpp

```diff
@@ -55,6 +58,35 @@ FailureOr<Version> Version::fromString(llvm::StringRef versionRef) {
   return Version(versionArr[0], versionArr[1], versionArr[2]);
 }
 
+FailureOr<int64_t> Version::getBytecodeFormatVersion(const Version& version) {
+  // Fail if requested version is not in supported versions.
+  if (vhlo::Version::getCurrentVersion() < version ||
+      version < vhlo::Version::getMinimumVersion()) {
+    return failure();
+  }
+
+  // This map is updated on all bytecode format increments.
+  // Updating to use a newer bytecode format requires a new minor version
+  // which must be added to this map.
+  const static std::vector<std::pair<Version, int64_t>> formatVersionMap{
+      // All versions greater than first entry use version of first entry.
+      {Version(0, 10, 0), 1},
+      {Version(0, 9, 0), 0},
```

Comment:
I think that in practice most or even all StableHLO portable artifacts with targetVersion = 0.9.0 have bytecodeVersion = 1?

Reply:
If a user requests "bytecode that can run at tagged release v0.9.0", and this API returns `1`, then it will generate bytecode that cannot run at tagged release v0.9.0.

A better way of describing the current situation is "We don't have users running tagged release v0.9.0", which can be deprecated since v0.10.0 was tagged on Apr 20. In the near future we should figure out a deprecation process and deprecate that release as it is no longer in the compatibility window.